calculations/temperature_calculations.py

```diff
@@ -1,4 +1,57 @@
 class TemperatureCalculator:
+    
+    def compile_data_for_yearly_calculations(self, weather_readings):
+        minimum_temperature_every_month = {}
+        maximum_temperature_every_month = {}
+        maximum_humidity_every_month = {}
+        
+        for year_number, year_values in weather_readings.items():
+            minimum_temperature = []
+            maximum_temperature = []
+            maximum_humidity = []
+            
+            for month_name, month_values in year_values.items():
+                minimum_temperature.append(
+                    min(month_values, key=lambda x: int(x['Min TemperatureC'])))
```

Comment:
We can use list comprehension here.

Reply:
list comprehension cannot be used because of dictionary.